### PR TITLE
Context Distiller

### PR DESCRIPTION
**Pull request by herlo**
The Context Distiller takes results data from provisioning resources, and distills it down into json data based upon fields that are set in the linchpin.constants. To enable this feature, the lp.distill_data flag needs to be set in linchpin.conf.

### CODE REVIEW

File: linchpin/__init__.py [hunk 1]
```diff
@@ -707,6 +705,69 @@ def do_action(self, provision_data, action='up', run_id=None, tx_id=None):
         return (return_code, lp_data)
 
 
+    def get_run_data(self, tx_id, fields, targets=()):
+        """
+        Returns the RunDB for data from a specified field given either a run_id
+        or tx_id. The fields consist of the major sections in the RunDB (target
```
Comment (p3ck) [inline linchpin/__init__.py]: Is this correct?  it says either a run_id or tx_id but I only see tx_id passed in to the method.

File: linchpin/cli/__init__.py [hunk 2]
```diff
@@ -173,6 +174,99 @@ def _get_data_path(self):
...
+            try:
+                for res in resources:
+                    for k, v in fields.iteritems():
+                        res_data = {}
+                        if not v:
+                            dd = dist_data.get(target, {})
+                            if len(dd):
+                                for items in dd:
+                                    if k in items.keys():
+                                        items[k].extend(res.get(k))
```
Comment (p3ck) [inline linchpin/cli/__init__.py]: Can you explain what this loop is doing?  Seems a little odd to me to be modifying the dictionary you are iterating over.

**Comment by herlo** [comment]
@p3ck this passed and could use a review. Thanks.

File: linchpin/cli/__init__.py (hunk 2 above)
Comment (herlo) [reply linchpin/cli/__init__.py]: Sure. There are actually two dictionaries here. The first pulls out the fields which need to be grabbed from the particular provider. The second matches those items and  puts them into the dictionary.

File: linchpin/__init__.py (hunk 1 above)
Comment (herlo) [reply linchpin/__init__.py]: tx_id holds the run_ids for a particular run (or transaction). To be able to get the data, the tx_id is used to get the run_ids. In turn, the resource data is pulled by the run_id into a returnable format.

Comment (herlo) [reply linchpin/__init__.py]: I see what you mean. I missed the point. The docs just need to remove the reference to the run_id. I'll get that fixed in the next update coming shortly.

**Comment by herlo** [comment]
Ignore the above comment. I was using the mobile version and it formats things a little differently.

File: linchpin/cli/__init__.py (hunk 2 above)
Comment (p3ck) [reply linchpin/cli/__init__.py]: ok - I get it now.

**Comment by herlo** [comment]
[merge]

**Pull request merged**